src/engine/basics/settings.hxx

```diff
@@ -40,7 +40,7 @@ public:
   SettingsStruct settings;
 
 private:
-  std::string settingsFileName = "resources/settings.json";
+  std::string m_settingsFileName = "resources/settings.json";
```

Comment:
Can user change the value of this variable?

Reply:
No, imho the location of the settings file should be hardcoded and not change. Do you agree?

Reply:
If so, what about changing it to a global constant and adding `constexpr` keyword?

Reply:
done

Reply:
I store global constants in another header file such as Constants.hpp. I wish you refer to that.
Also, I think that `SETTINGS_FILE_NAME` is better. `SETTINGSFILENAME` is hard to read.

Reply:
done :)

Reply:
also renamed the constant in settings.cxx

Reply:
I confirmed it is fixed.